From alan@linux.intel.com  Mon Oct 26 14:55:59 2009
From: Alan Cox <alan@linux.intel.com>
Date: Tue, 06 Oct 2009 15:49:04 +0100
Subject: Staging: et131x: Clean up the receive arrays
To: greg@kroah.com
Message-ID: <20091006144902.8604.7635.stgit@localhost.localdomain>


We don't use them for anything having stripped out the debug gunge in
the original driver.

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/et131x/et1310_rx.c |   19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

--- a/drivers/staging/et131x/et1310_rx.c
+++ b/drivers/staging/et131x/et1310_rx.c
@@ -1076,10 +1076,7 @@ void et131x_reset_recv(struct et131x_ada
 void et131x_handle_recv_interrupt(struct et131x_adapter *etdev)
 {
 	PMP_RFD rfd = NULL;
-	struct sk_buff *packets[NUM_PACKETS_HANDLED];
-	PMP_RFD freed[NUM_PACKETS_HANDLED];
 	u32 count = 0;
-	u32 nfree = 0;
 	bool done = true;
 
 	/* Process up to available RFD's */
@@ -1110,24 +1107,10 @@ void et131x_handle_recv_interrupt(struct
 		etdev->Stats.ipackets++;
 
 		/* Set the status on the packet, either resources or success */
-		if (etdev->RxRing.nReadyRecv >= RFD_LOW_WATER_MARK) {
-			/* Put this RFD on the pending list
-			 *
-			 * NOTE: nic_rx_pkts() above is already returning the
-			 * RFD to the RecvList, so don't additionally do that
-			 * here.
-			 * Besides, we don't really need (at this point) the
-			 * pending list anyway.
-			 */
-		} else {
-			freed[nfree] = rfd;
-			nfree++;
-
+		if (etdev->RxRing.nReadyRecv < RFD_LOW_WATER_MARK) {
 			dev_warn(&etdev->pdev->dev,
 				    "RFD's are running out\n");
 		}
-
-		packets[count] = rfd->Packet;
 		count++;
 	}
 
